api/app/src/external/commonwell/document/document-query.ts

```diff
@@ -90,23 +101,101 @@ async function internalGetDocuments({
     throw err;
   }
 
-  const documents: DocumentWithLocation[] = docs.entry
+  const documents: Document[] = docs.entry
     ? docs.entry.flatMap(d =>
-        d.content?.masterIdentifier?.value && d.content?.location
-          ? {
-              id: d.content.masterIdentifier.value,
-              fileName: getFileName(patient, d),
-              description: d.content.description,
-              type: d.content.type,
-              status: d.content.status,
-              location: d.content.location,
-              indexed: d.content.indexed,
-              mimeType: d.content.mimeType,
-              size: d.content.size, // bytes
-              raw: d,
-            }
+        d.id && d.content && d.content.location
+          ? { id: d.id, content: { location: d.content.location, ...d.content } }
           : []
       )
     : [];
   return documents;
 }
+
+async function dowloadAnduploadDocsToS3({
+  patient,
+  facilityId,
+  documents,
+}: {
+  patient: Patient;
+  facilityId: string;
+  documents: Document[];
+}): Promise<DocumentWithFilename[]> {
+  const uploadStream = (key: string) => {
+    const pass = new stream.PassThrough();
+    const removePeriods = key.split(".").join("");
+
+    return {
+      writeStream: pass,
+      promise: s3client
+        .upload({ Bucket: Config.getMedicalDocumentsBucketName(), Key: removePeriods, Body: pass })
+        .promise(),
+    };
+  };
+
+  const s3Refs = await Promise.allSettled(
+    documents.map(async doc => {
+      if (doc.content?.masterIdentifier?.value) {
+        // TEMP KEY
+        const { writeStream, promise } = uploadStream(doc.content?.masterIdentifier?.value);
+
+        await downloadDocument({
+          cxId: patient.cxId,
+          patientId: patient.id,
+          facilityId: facilityId,
+          location: doc.content?.location ? doc.content.location : "",
+          stream: writeStream,
+        });
+
+        const data = await promise;
+
+        return {
+          ...doc,
+          content: {
+            ...doc.content,
+            location: data.Location,
+          },
+          fileName: data.Key,
+        };
+      }
+
+      return undefined;
+    })
+  );
+
+  const docsNewLocation: DocumentWithFilename[] = s3Refs.flatMap(ref =>
+    ref.status === "fulfilled" && ref.value ? ref.value : []
+  );
+
+  s3Refs.forEach(ref => {
+    if (ref.status === "rejected") {
+      capture.error(ref.reason, {
+        extra: {
+          context: `s3.documentUpload`,
+        },
+      });
+    }
+  });
```

Comment:
Could this get crazy if 100s of documents fail? Should we bundle these up in a single capture?

@leite08 - any thoughts here would be good, I'm not familiar with how Sentry handles these situations

Reply:
I can count them and then do a generic message so  its like 

extra: {
      context: `s3.documentUpload`,
      count: 12
}